src/NCronJob/Scheduler/JobWorker.cs

```diff
@@ -40,13 +40,14 @@ public JobWorker(
 
     public async Task WorkerAsync(string queueName, CancellationToken cancellationToken)
     {
-        var jobQueue = jobQueueManager.GetOrAddQueue(queueName);
         var concurrencyLimit = registry.GetJobTypeConcurrencyLimit(queueName);
         var semaphore = jobQueueManager.GetOrAddSemaphore(queueName, concurrencyLimit);
         var runningTasks = new List<Task>();
 
         while (!cancellationToken.IsCancellationRequested)
         {
+            var jobQueue = jobQueueManager.GetOrAddQueue(queueName);
```

Comment:
This change allows a "waiting" job to get an always "fresh" queue. Indeed, the queue may have been dropped and recreated behind the scene (through a Disable/Enable job operation for instance)

Reply:
<3